coordinator/coordinator.go

```diff
@@ -113,10 +110,47 @@ func (c *Coordinator) AwaitPendingSource(sourceNameVariant metadata.NameVariant)
 		if sourceStatus == metadata.READY {
 			return source, nil
 		}
-		elapsed = time.Since(start)
 		time.Sleep(1 * time.Second)
 	}
-	return nil, fmt.Errorf("waited too long for source to become ready")
+	return c.Metadata.GetSourceVariant(context.Background(), sourceNameVariant)
+}
+
+func (c *Coordinator) AwaitPendingFeature(featureNameVariant metadata.NameVariant) (*metadata.FeatureVariant, error) {
+	featureStatus := metadata.PENDING
+	for featureStatus != metadata.READY {
+		feature, err := c.Metadata.GetFeatureVariant(context.Background(), featureNameVariant)
+		if err != nil {
+			return nil, err
+		}
+		featureStatus := feature.Status()
+		if featureStatus == metadata.FAILED {
+			return nil, fmt.Errorf("source of feature not ready: name: %s, variant: %s", featureNameVariant.Name, featureNameVariant.Variant)
```

Comment:
Can you change error message to "feature registration failed: ......". Same with the label and source. Want it to be explicit that its in a failed state